tests/cpp/test_multidevice_sharding.cpp

```diff
@@ -40,6 +40,30 @@ TEST_F(NVFuserTest, TestIsSharded) {
   EXPECT_ANY_THROW(isSharded(c));
 }
 
+TEST_F(NVFuserTest, TestPropagateSharding) {
+  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
```

Comment:
```suggestion
  Fusion fusion;
```
I don't see a need to use a smart pointer.